### PR TITLE
Define method for signing a deal proposal

### PR DESCRIPTION
**Pull request by magik6k**

### CODE REVIEW

File: actors/crypto/signature.go [hunk 1]
```diff
@@ -13,3 +17,9 @@ type Signature struct {
 	Type SigType
 	Data []byte
 }
+
+type SignFunc = func(context.Context, []byte) (*Signature, error)
+
+type Signable interface {
+	Sign(context.Context, SignFunc) error
```
Comment (magik6k) [inline actors/crypto/signature.go]: Note: lotus needs this context to be able to call wallet over RPC

File: actors/builtin/market/deal.go [hunk 2]
```diff
@@ -1,8 +1,12 @@
 package market
 
 import (
+	"bytes"
+	"context"
+
 	addr "github.com/filecoin-project/go-address"
 	cid "github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
```
Comment (anorth) [inline actors/builtin/market/deal.go]: use `pkg/errors` for consistency with the rest of the project

File: actors/builtin/market/deal.go [hunk 3]
```diff
@@ -22,7 +26,7 @@ type DealProposal struct {
 	PieceSize       abi.PaddedPieceSize
 	Client          addr.Address
 	Provider        addr.Address
-	ClientSignature acrypto.Signature
+	ClientSignature *acrypto.Signature
```
Comment (anorth) [inline actors/builtin/market/deal.go]: I literally just confirmed with @whyrusleeping that non-pointers is appropriate for all the nested structures. What motivates this change? Why not just use the zero value?

File: actors/crypto/signature.go [hunk 4]
```diff
@@ -1,5 +1,9 @@
 package crypto
 
+import (
+	"context"
+)
+
 type SigType int64
```
Comment (anorth) [inline actors/crypto/signature.go]: Suggestion: use `0` for SigTypeUnknown so it's the ZeroValue?

**Pull request closed without merging**